sdk/src/activity_context.rs

```diff
@@ -0,0 +1,212 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{Duration as StdDuration, SystemTime};
+
+use prost_types::{Duration, Timestamp};
+use temporal_sdk_core_api::Worker;
+use temporal_sdk_core_protos::coresdk::{
+    activity_task,
+    common::{Payload, RetryPolicy, WorkflowExecution},
+    ActivityHeartbeat,
+};
+use tokio_util::sync::CancellationToken;
+
+/// Used within activities to get info, heartbeat management etc.
+#[derive(Clone)]
+#[allow(dead_code)]
+pub struct ActContext {
+    pub(crate) worker: Arc<dyn Worker>,
+    pub(crate) cancellation_token: CancellationToken,
+    pub(crate) input: Vec<Payload>,
+    pub(crate) heartbeat_details: Vec<Payload>,
+    /// #NOTE future usage?
+    pub(crate) header_fields: HashMap<String, Payload>,
+    pub(crate) info: ActivityInfo,
+}
+
+#[derive(Clone)]
+pub struct ActivityInfo {
+    pub task_token: Vec<u8>,
+    pub workflow_type: String,
+    pub workflow_namespace: String,
+    pub workflow_execution: Option<WorkflowExecution>,
+    pub activity_id: String,
+    pub activity_type: String,
+    pub task_queue: String,
+    pub heartbeat_timeout: Option<Duration>,
+    /// Time of activity scheduled by a workflow
+    pub scheduled_time: Option<Timestamp>,
+    /// Time of activity start
+    pub started_time: Option<Timestamp>,
+    /// Time of activity timeout
+    pub deadline: Option<Timestamp>,
```

Comment:
Sorry I didn't notice this earlier - but lets change these to stdlib types. They're much easier for users to work with than the prost types.

If they cannot be converted the context constructor can bubble up an error.